lib/ember-test-helpers/test-module.js

```diff
@@ -269,6 +271,36 @@ export default Klass.extend({
     };
   },
 
+  /*
+    Builds a version of the passed in context that contains deprecation warnings
+    for accessing properties that exist on the module.
+  */
+  _buildDeprecatedContext: function(module, context) {
+    var deprecatedContext = Object.create(context);
```

Comment:
Wait, sorry, I misread.  This is only being called when needed (because `_contextualizeCallback` is called only when needed), but we can still cache the deprecated context (instead of building a new one for every function that is in `options`).